dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -239,6 +241,36 @@ public static Tool convertEntryToTool(Entry<?, ?> container, DockstoreWebservice
         return tool;
     }
 
+    /**
+     * Construct the image_name for ImageData
+     *
+     * @return The full image name
+     */
+    private static String constructImageName(final Image image) {
+        DockerSpecifier specifier = image.getSpecifier();
+        Registry registry = image.getImageRegistry();
+        String fullRepositoryName = image.getRepository();
+
+        if (registry != Registry.DOCKER_HUB) {
+            // If the registry is not Docker Hub (ex: Quay), prepend the image registry docker path
+            fullRepositoryName = String.join("/", image.getImageRegistry().getDockerPath(), fullRepositoryName);
+        }
+
+        // Check if specifier is null because tool images don't have the DockerSpecifier set properly yet. For now, if it's null, assume that it's a tag
+        // TODO: Remove null check once the DockerSpecifier for tool images are set properly
+        if (specifier == null || specifier == DockerSpecifier.TAG) {
+            return String.join(":", fullRepositoryName, image.getTag());
+        } else if (specifier == DockerSpecifier.DIGEST) {
+            // The image's checksum is the image's digest
+            String imageDigest = image.getChecksums().get(0).toString();
```

Comment:
This could be overkill to handle this, but if we ever add support for other checksum types, this code may not work -- you'll need to check the checksum type.